src/instantsend/instantsend.h

```diff
@@ -135,7 +135,7 @@ class CInstantSendManager final : public instantsend::InstantSendSignerParent
     /* Helpers for communications between CInstantSendManager & NetInstantSend */
     // This helper returns up to 32 pending locks and remove them from queue of pending
     [[nodiscard]] instantsend::PendingState FetchPendingLocks() EXCLUSIVE_LOCKS_REQUIRED(!cs_pendingLocks);
-    void EnqueueInstantSendLock(NodeId from, const uint256& hash, const std::shared_ptr<instantsend::InstantSendLock>& islock)
+    void EnqueueInstantSendLock(NodeId from, const uint256& hash, std::shared_ptr<instantsend::InstantSendLock>&& islock)
```

Comment:
take it by value; not by &&, imo. then on https://github.com/dashpay/dash/pull/6959/commits/cab3b27028ba9c423e5dbef23d9190f5a9fdcf92#diff-ad0e0b034e03426cef39d5c87fbe3fde42546c13072c6d89040dbb6fc7f45ff1R89 move it again

Reply:
@coderabbitai justify both options: pass std::shared_ptr as && and as a value; benefits and downsides if any.

Reply:
I love that you referenced "This is the approach recommended by C++ experts like Herb Sutter in his "Back to Basics" talks." :D